xc/device.ml

```diff
@@ -1143,18 +1143,67 @@ module PCI = struct
     (* Sort into the order the devices were plugged *)
     List.sort (fun a b -> compare (fst a) (fst b)) pairs
 
-  let add ~xs pcidevs domid =
+  let _pci_add ~xc ~xs pcidev domid slot =
```

Comment:
I think the slot here was meant to be a guest one (judging from the caller), but is mixed with a host one below.
Using the `Pci.address` type as @robhoes suggested might be better, so it is obvious what is guest and what is host.

Reply:
I've made some changes from other comments, and now I'm not sure where you're referring to. I'll push my changes, and perhaps ask you offline / again